test/unit/state_test.go

```diff
@@ -0,0 +1,42 @@
+package unit
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"gopkg.in/yaml.v2"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
+)
+
+var _ = Describe("State", func() {
+	var (
+		obtained nmstatev1.NodeNetworkState
+		oneIf    = `apiVersion: nmstate.io/v1
+kind: NodeNetworkState
+metadata:
+  name: node01
+spec:
+  managed: true
+  nodeName: node01
+  desiredState:
+    interfaces:
+      - name: eth1
+        type: ethernet
+        state: up
+`
+	)
+	BeforeEach(func() {
+		err := yaml.Unmarshal([]byte(oneIf), &obtained)
```

Comment:
I think this could be done more drastically.

```go
var _ = Describe("State", func() {
	obtained nmstatev1.NodeNetworkState
	err := yaml.Unmarshal([]byte(`
apiVersion: nmstate.io/v1
kind: NodeNetworkState
metadata:
  name: node01
spec:
  managed: true
  nodeName: node01
  desiredState:
    interfaces:
      - name: eth1
        type: ethernet
        state: up
`), &obtained)
	if err != nil {
		panic(err)
	}
```

Or even better, introduce a helper function for that, we will transform manifest string to the structure everywhere.

Also, it should be defined under context. We use this manifest for one specific purpose now. If we later see it can be reuse, we can move it higher. But let's keep it close to its usage to avoid junkyard.

Reply:
I have changed it a lot, let review it again.